test/testdata/parser/error_recovery/singleton_method_literal.rb

```diff
@@ -0,0 +1,8 @@
+# typed: true
+
+def (10).test()
+   # ^^ error: cannot define a singleton method for a literal
```

Comment:
Just to be sure, is this reported at typed false? I don’t want sorbet to implicitly think that there’s a singleton method in the enclosing scope silently.

Reply:
Yes, that's the `SingletonLiteral` diagnostic being raised.